objects/SetupChecker/script.lua

```diff
@@ -1977,65 +1998,46 @@ function getNewSpirit(tags, complexities)
         return nil
     end
     local aspect = sourceSpirit.call("RandomAspect", {obj = spirit})
-    spiritChoices[spirit.getName()] = {guid=spirit.guid, aspect=aspect}
+    spiritChoices[spirit.getName()] = {guid=spirit.guid}
     spiritChoicesLength = spiritChoicesLength + 1
     return spirit, aspect
 end
-function pickSpirit1(obj, color)
-    if Global.getTable("playerTables")[color] ~= obj then return end
-    pickSpirit(obj, 0, color)
-end
-function pickSpirit2(obj, color)
-    if Global.getTable("playerTables")[color] ~= obj then return end
-    pickSpirit(obj, 1, color)
-end
-function pickSpirit3(obj, color)
-    if Global.getTable("playerTables")[color] ~= obj then return end
-    pickSpirit(obj, 2, color)
-end
-function pickSpirit4(obj, color)
-    if Global.getTable("playerTables")[color] ~= obj then return end
-    pickSpirit(obj, 3, color)
-end
-function replaceSpirit(obj, index, color)
+function replaceSpirit(obj, oldSpirit, player)
     local tags = getSpiritTags()
     if tags == nil then
-        Player[color].broadcast("You have no expansions selected", Color.Red)
+        player.broadcast("You have no expansions selected", Color.Red)
         return
     end
     local complexities = getSpiritComplexities()
     if complexities == nil then
-        Player[color].broadcast("You have no complexities selected", Color.Red)
+        player.broadcast("You have no complexities selected", Color.Red)
         return
     end
-    local spirit = getNewSpirit(tags, complexities)
+    local spirit, aspect = getNewSpirit(tags, complexities)
+    local choices = getGainSpiritChoices(obj)
+    for _, choice in pairs(choices) do
+        if choice.spirit == oldSpirit then
+            choice.spirit = spirit and spirit.getName()
+            choice.aspect = aspect
```

Comment:
there should be a break after this

Reply:
Yeah, a break here wouldn't hurt, though the code is still correct without it.